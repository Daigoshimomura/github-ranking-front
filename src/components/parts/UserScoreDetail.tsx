import React from "react";
import { UserDetailRes } from "../../external/data/UserDetailRes";
import {
  TableContainer,
  Table,
  TableHead,
  TableRow,
  TableCell,
  Paper,
  TableBody
} from "@material-ui/core";

interface UserScoreDetailPorps {
  userDetail: UserDetailRes;
}

export class UserScoreDetail extends React.Component<UserScoreDetailPorps, {}> {
  render() {
    return (
      <TableContainer component={Paper}>
        <Table aria-label="simple table">
          <TableHead>
            <TableRow>
              <TableCell>Detail Score</TableCell>
              <TableCell align="right">followers</TableCell>
              <TableCell align="right">issues</TableCell>
              <TableCell align="right">pullRequests</TableCell>
              <TableCell align="right">repositories</TableCell>
              <TableCell align="right">forks</TableCell>
              <TableCell align="right">stargazer</TableCell>
              <TableCell align="right">watchers</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            <TableRow>
              <TableCell component="th" scope="row"></TableCell>
              <TableCell align="right">
                {this.props.userDetail.followersCount}
              </TableCell>
              <TableCell align="right">
                {this.props.userDetail.issuesCount}
              </TableCell>
              <TableCell align="right">
                {this.props.userDetail.pullRequestCount}
              </TableCell>
              <TableCell align="right">
                {this.props.userDetail.repositoriesCount}
              </TableCell>
              <TableCell align="right">
                {this.props.userDetail.forksCountTotal}
              </TableCell>
              <TableCell align="right">
                {this.props.userDetail.stargazerCountTotal}
              </TableCell>
              <TableCell align="right">
                {this.props.userDetail.watchersCountTotal}
              </TableCell>
            </TableRow>
          </TableBody>
        </Table>
      </TableContainer>
    );
  }
}
