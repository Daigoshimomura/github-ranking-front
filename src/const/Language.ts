export const GENERAL = "General";
export const JAVASCRIPT = "JavaScript";
export const PYTHON = "Python";
export const JAVA = "Java";
export const CSHARP = "C#";
export const PHP = "PHP";
export const C_PLUS = "C++";
export const RUBY = "Ruby";
export const GO = "Go";
