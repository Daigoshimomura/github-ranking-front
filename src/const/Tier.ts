export const CHALLENGER = "Challenger";
export const MASTER = "Master";
export const DIAMOND = "Diamond";
export const PRATINUM = "Pratinum";
export const GOLD = "Gold";
export const SILVER = "Silver";
export const BRONZE = "Bronze";
