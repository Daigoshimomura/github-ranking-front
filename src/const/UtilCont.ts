export const NOTFOUND = "未発見";
export const UNSEARCHED = "未検索";
