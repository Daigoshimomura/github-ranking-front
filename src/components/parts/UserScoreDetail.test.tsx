import { mount } from "enzyme";
import React from "react";
import { UserScoreDetail } from "./UserScoreDetail";
import { TableContainer } from "@material-ui/core";
import { UNSEARCHED } from "../../const/UtilCont";
import renderer from "react-test-renderer";

const userDetail = {
  userName: "userName",
  userId: UNSEARCHED,
  avatarUrl: "",
  githubUrl: "",
  tier: "",
  rank: 0,
  score: 0,
  currentNumber: 0,
  followersCount: 0,
  issuesCount: 0,
  pullRequestCount: 0,
  repositoriesCount: 0,
  forksCountTotal: 0,
  stargazerCountTotal: 0,
  watchersCountTotal: 0,
  mainLanguage: "",
  lastupdateDate: new Date()
};

describe("UserScoreDetail", () => {
  it("snapshot", () => {
    const tree = renderer
      .create(<UserScoreDetail userDetail={userDetail} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("レンダリングの確認", () => {
    const wrapper = mount(<UserScoreDetail userDetail={userDetail} />);
    expect(wrapper.find(TableContainer).length).toBe(1);
  });
  it("propsをちゃんと渡せてることの確認", () => {
    const wrapper = mount(<UserScoreDetail userDetail={userDetail} />);
    expect(wrapper.prop("userDetail").userName).toBe("userName");
  });
});
