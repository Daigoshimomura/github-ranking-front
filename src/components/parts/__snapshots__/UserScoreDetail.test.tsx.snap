// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserScoreDetail snapshot 1`] = `
<div
  className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <table
    aria-label="simple table"
    className="MuiTable-root"
  >
    <thead
      className="MuiTableHead-root"
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Detail Score
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          followers
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          issues
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          pullRequests
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          repositories
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          forks
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          stargazer
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          watchers
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
    >
      <tr
        className="MuiTableRow-root"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
          scope="row"
        />
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          0
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          0
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          0
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          0
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          0
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          0
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          0
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
